ui/components/multichain/account-list-menu/account-list-menu.js

```diff
@@ -116,6 +118,29 @@ export const AccountListMenu = ({
     }
   }
 
+  if (process.env.NETWORK_ACCOUNT_DND && pinnedAccounts.length > 0) {
```

Comment:
The contents of these blocks should be in a selector, IMO.  Doing data management/massaging in components can be inefficient.  Also, moving all of this to a selector will make writing a test much easier.

Reply:
Nice shout, Updated here https://github.com/MetaMask/metamask-extension/pull/21865/commits/8673c6a9d9cdeccc5ca2a5b45b32a171b9dff4cd